frontend/app.tsx

```diff
@@ -340,6 +276,64 @@ function reducer(
   return state;
 }
 
+function diffReducer(state: State, diff: Diff): State {
+  if (diff.length === 0) {
+    return state;
+  }
+  const newAllIssuesMap = new Map(state.allIssuesMap);
+  let newViewIssueCount = state.viewIssueCount;
+  const newFilteredIssues = [...state.filteredIssues];
+  const orderIteratee = partial(getOrderValue, state.issueOrder);
+
+  function add(key: string, newValue: ReadonlyJSONValue) {
+    const newIssue = issueFromKeyAndValue(key, newValue);
+    newAllIssuesMap.set(key, newIssue);
+    if (state.filters.viewFilter(newIssue)) {
+      newViewIssueCount++;
+    }
+    if (state.filters.issuesFilter(newIssue)) {
+      newFilteredIssues.splice(
+        sortedIndexBy(newFilteredIssues, newIssue, orderIteratee),
+        0,
+        newIssue
+      );
+    }
+  }
+  function del(key: string, oldValue: ReadonlyJSONValue) {
+    const oldIssue = issueFromKeyAndValue(key, oldValue);
+    const index = sortedIndexBy(newFilteredIssues, oldIssue, orderIteratee);
+    newAllIssuesMap.delete(key);
+    if (state.filters.viewFilter(oldIssue)) {
+      newViewIssueCount--;
+    }
+    if (newFilteredIssues[index]?.id === oldIssue.id) {
+      newFilteredIssues.splice(index, 1);
+    }
+  }
+  for (const diffOp of diff) {
+    switch (diffOp.op) {
+      case "add": {
+        add(diffOp.key, diffOp.newValue);
+        break;
+      }
+      case "del": {
+        del(diffOp.key, diffOp.oldValue);
+        break;
+      }
+      case "change": {
+        del(diffOp.key, diffOp.oldValue);
```

Comment:
Don't you have to `add()` in this case too?

Reply:
correct, done.